### DISCUSSION TITLE
Launcher for osu!

### DISCUSSION THREAD

**Discussion by Akornzzz** [ppy/osu#24865] [Ideas] [answered]
I think that there should be some sort of launcher for osu. Like for when you don’t want to play you can have the launcher open and have immediate access to download and preview beatmaps, message your friends, get notifications, talk in the community, talk through osu chat, launch the game etc. It’d be kinda like the website but with more to it.
 
It could work something like this:
-You could launch the game like normal and close the game like normal.
-You could launch the game through the launcher and close the game like normal which would minimize to the launcher.
 
It just seems like it’d be convenient. Although you've probably thought about this before.

**Comment by peppy** [comment]
Why not just leave the game open? It has everything you've asked for..

**Reply by nana123126** [reply]
-- It takes up too much resources and attention --

Games in the background will consume memory and continuously consume CPU and GPU resources, especially after playing a game (perhaps, I only remember consuming 1.2GB of memory in the background before, which is not the point).

Perhaps we should optimize the backend performance, or design an interface in the game that does not require loading a complex beatmap list, resulting in excessive performance consumption. The main menu may be a method, but the premise for doing so is at least to optimize performance and resources during the main menu.

I think we can add another feature: 'Full screen only in song lists and games', which allows the game to quickly switch to other windows on the main menu and meets all the requirements of' akkyxz '. But this may be a special requirement, and I don't think it will be used by many people, making it a bit bulky on the settings list, or perhaps a brand new settings panel is needed? That's another big project.

It may not seem suitable for mobile platforms, but mobile platforms themselves do not require this requirement.

**Reply by nana123126** [reply]
> I only remember consuming 1.2GB of memory in the background before

Ah, I took a look and actually saw that it consumed 1.5GB of graphics memory, in main menu

**Reply by peppy** [reply, marked as answer]
I'm not sure how you got osu! to use 1.5gb GPU memory. Probably a faulty skin.

Anyways, this is the direction we're going. Optimisations will continue.
